File: core/FitAR/FitAR.API/Models/ModelBase.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace FitAR.Web.API.Models
{
  public class ModelBase
  {
    public bool hasError { get; set; } = false;
    public string errorMessage { get; set; } = string.Empty;

    public void GenerateError(string err)
    {
      this.hasError = true;
      this.errorMessage = err;
    }

  }
}

File: core/android/app/src/main/java/com/aco/fitar/api/ModelBase.kt
package com.aco.fitar.api

import kotlinx.serialization.Serializable
import java.lang.reflect.ParameterizedType

@Serializable
abstract class ModelBase {

}

@Serializable
abstract  class ModelResponseBase{
    public var errorMessage:String = "";
    public var hasError:Boolean = false;

    companion object{
        public fun<T> generateError(err:String, type:Class<T>):T where T: ModelResponseBase{
            var respones = type.newInstance();
            respones.hasError = true;
            respones.errorMessage = err;
            return respones;
        }
    }
}

class DefaultModelResponse : ModelResponseBase(){ }